---
params:
  title: "Seurat Markers"
  
  # Use a SingleCellExperiment coerced from Seurat.
  sce_file: "data/YYYY-MM-DD/sce.rds"
  
  markers_file: "data/YYYY-MM-DD/markers.rds"
  cell_type_markers_file: NULL
  organism: "Homo sapiens"
  
  # Use dark mode theme (`theme_midnight()`) for plots.
  dark: FALSE
  
  data_dir: !r file.path("rds", Sys.Date())
  results_dir: !r file.path("results", Sys.Date(), "markers")

title: "`r params$title`"
author: "`r getOption('author')`"
date: "`r Sys.Date()`"
bibliography: bibliography.bib
---

```{r setup, cache=FALSE, message=FALSE}
# Last modified 2019-04-17

library(Seurat)
library(pointillism)
prepareTemplate(package = "pointillism")
source("_setup.R")

# Create directories.
dir.create(params$results_dir, recursive = TRUE, showWarnings = FALSE)

# knitr arguments (for `rmarkdown::render()` looping)
# > opts_chunk$set(
# >     cache.path = paste(
# >         sce_name,
# >         "markers",
# >         "cache/",
# >         sep = "_"
# >     ),
# >     fig.path = paste(
# >         sce_name,
# >         "markers",
# >         "files/",
# >         sep = "_"
# >     )
# > )
```

```{r header, child="_header.Rmd"}
```

# Load SingleCellExperiment object

Use a `SingleCellExperiment` object coerced from `Seurat` as input.
This is more memory efficient than using a `Seurat` object directly.

```{r load-sce}
stopifnot(file.exists(params$sce_file))
if (fileExt(params$sce_file) == "rds") {
    sce <- readRDS(file = params$sce_file)
    name <- basenameSansExt(params$sce_file)
} else if (fileExt(params$sce_file) == "rda") {
    name <- load(file = params$sce_file)
    sce <- get(x = name, inherits = FALSE)
}
stopifnot(
    is(sce, "SingleCellExperiment"),
    is.character(name)
)
invisible(validObject(sce))
print(sce)

# Require that the object contains logcounts.
stopifnot("logcounts" %in% assayNames(sce))
stopifnot(!is.null(logcounts(sce)))
```

# Differential expression

Seurat can help you find markers that define clusters via differential expression. By default, it identifes positive and negative markers of a single cluster (specified in `ident.1`), compared to all other cells. `FindAllMarkers()` automates this process for all clusters, but you can also test groups of clusters vs. each other, or against all cells.

The `min.pct` argument requires a gene to be detected at a minimum percentage in either of the two groups of cells, and the `thresh.test` argument requires a gene to be differentially expressed (on average) by some amount between the two groups. You can set both of these to 0, but with a dramatic increase in time - since this will test a large number of genes that are unlikely to be highly discriminatory. As another option to speed up these computations, `max.cells.per.ident` can be set. This will downsample each identity class to have no more cells than whatever this is set to. While there is generally going to be a loss in power, the speed increases can be significiant and the most highly differentially expressed genes will likely still rise to the top.

Seurat has multiple tests for differential expression which can be set with the `test.use` parameter. Currently it defaults to Wilcoxon rank sum test ("`wilcox`"), which tends to overreport significant markers. Instead we currently recommend using MAST (Finak et al, Genome Biology, 2015) or DESeq2 (Love et al, Genome Biology, 2014).

It is recommended to run this calculation as a seprate R script outside of R Markdown, since it can take a long time for large datasets.

First, import the unmodified Seurat `FindAllMarkers()` `data.frame` return.

```{r load-seurat-markers}
markers_name <- load(params$markers_file)
markers <- get(markers_name, inherits = FALSE)
stopifnot(is.data.frame(markers))

# Keep a backup of the original data.frame.
markers_original <- markers

# Default to using our internal cell type markers, managed on Google Sheets.
# File on issue on the bcbioSingleCell GitHub repo if this list needs to get
# updated or the desired organism isn't yet supported.
if (!is.null(params$cell_type_markers_file)) {
    gene2symbol <- gene2symbol(sce)
    cell_type_markers <- readCellTypeMarkers(
        file = params$cell_type_markers_file,
        gene2symbol = gene2symbol
    )
} else {
    stopifnot(!is.null(params$organism))
    cell_type_markers <- pointillism::cell_type_markers %>%
        .[[camel(params$organism)]]
}
stopifnot(is.data.frame(cell_type_markers))
assignAndSaveData(
    name = paste(sce_name, "cell_type_markers", sep = "_"),
    object = cell_type_markers,
    dir = params$data_dir
)
```

Here we are sanitizing the markers return, and adding additional metadata.

```{r sanitize-seurat-markers}
rowRanges <- rowRanges(seurat)
stopifnot("geneName" %in% colnames(mcols(rowRanges)))

# Sanitize the markers data.frame and include gene metadata.
all_markers <- sanitizeSeuratMarkers(
    data = all_markers,
    rowRanges = rowRanges
)
assignAndSaveData(
    name = paste(sce_name, "markers", "sanitized", sep = "_"),
    object = all_markers,
    dir = params$data_dir
)
write_csv(
    all_markers,
    path = file.path(
        params$results_dir,
        paste0(sce_name, "_markers_sanitized.csv.gz")
    )
)
```

# Top markers per cluster {.tabset}

```{r top-markers}
topMarkers(markers)
```

```{r plot-top-markers, results="asis"}
plotTopMarkers(
    object = sce,
    markers = markers,
    headerLevel = 2,
    dark = params$dark
)
```

# Known cell type markers {.tabset}

```{r known-markers-detected}
known_markers_detected <- knownMarkersDetected(
    all = all_markers,
    known = cell_type_markers
)
assignAndSaveData(
    name = paste(sce_name, "known_markers_detected", sep = "_"),
    object = known_markers_detected,
    dir = params$data_dir
)
known_markers_detected
```

```{r plot-known-markers-detected, results="asis"}
plotKnownMarkersDetected(
    object = sce,
    markers = known_markers_detected,
    headerLevel = 2,
    dark = params$dark
)
```

# Cell types per cluster {.tabset}

```{r cell-types-per-cluster}
cell_types_per_cluster <- cellTypesPerCluster(known_markers_detected)
assignAndSaveData(
    name = paste(sce_name, "cell_types_per_cluster", sep = "_"),
    object = cell_types_per_cluster,
    dir = params$data_dir
)
cell_types_per_cluster
```

```{r plot-cell-types-per-cluster, results="asis"}
plotCellTypesPerCluster(
    object = sce,
    markers = cell_types_per_cluster,
    headerLevel = 2,
    dark = params$dark
)
```

```{r footer, child="_footer.Rmd"}
```
