<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Differential expression — diffExp • pointillism</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Differential expression — diffExp" />
<meta property="og:description" content="Differential expression" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pointillism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../articles/pointillism.html">Get started</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://acidgenomics.com/">
    <span class="fas fa fas fa-vial fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/acidgenomics/pointillism/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Differential expression</h1>
    <small class="dont-index">Source: <a href='https://github.com/acidgenomics/pointillism/blob/master/R/AllGenerics.R'><code>R/AllGenerics.R</code></a>, <a href='https://github.com/acidgenomics/pointillism/blob/master/R/diffExp-methods.R'><code>R/diffExp-methods.R</code></a></small>
    <div class="hidden name"><code>diffExp.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Differential expression</p>
    </div>

    <pre class="usage"><span class='fu'>diffExp</span><span class='op'>(</span><span class='va'>object</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for SingleCellExperiment</span>
<span class='fu'>diffExp</span><span class='op'>(</span>
  <span class='va'>object</span>,
  <span class='va'>numerator</span>,
  <span class='va'>denominator</span>,
  caller <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"edgeR"</span>, <span class='st'>"DESeq2"</span><span class='op'>)</span>,
  minCells <span class='op'>=</span> <span class='fl'>2L</span>,
  minCellsPerGene <span class='op'>=</span> <span class='fl'>1L</span>,
  minCountsPerCell <span class='op'>=</span> <span class='fl'>1L</span>,
  BPPARAM <span class='op'>=</span> <span class='fu'>BiocParallel</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>bpparam</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span>

<span class='co'># S4 method for Seurat</span>
<span class='fu'>diffExp</span><span class='op'>(</span><span class='va'>object</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>numerator</th>
      <td><p><code>character</code>.
Cells to use in the numerator of the contrast (e.g. treatment).</p></td>
    </tr>
    <tr>
      <th>denominator</th>
      <td><p><code>character</code>.
Cells to use in the denominator of the contrast (e.g. control).</p></td>
    </tr>
    <tr>
      <th>caller</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Package to use for differential expression calling. Defaults to <code>"edgeR"</code>
(faster for large datasets) but <code>"DESeq2"</code> is also supported.</p></td>
    </tr>
    <tr>
      <th>minCells</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Minimum number of cells required to perform the differential expression
analysis.</p></td>
    </tr>
    <tr>
      <th>minCellsPerGene</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
The minimum number of cells where a gene is expressed, to pass low
expression filtering.</p></td>
    </tr>
    <tr>
      <th>minCountsPerCell</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Minimum number of counts per cell for a gene to pass low expression
filtering. The number of cells is defined by <code>minCellsPerGene</code>.</p></td>
    </tr>
    <tr>
      <th>BPPARAM</th>
      <td><p><code>bpparamClass</code>.
Back-end method to be used for computations.
See <code><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>bpparam</a></code> for details.
Currently only used by DESeq2 but not edgeR for calculations here.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Varies depending on the <code>caller</code> argument:</p><ul>
<li><p><code>caller = "edgeR"</code>: <code>DEGLRT</code>.</p></li>
<li><p><code>caller = "DESeq2"</code>: Unshrunken <code>DESeqResults</code>.
Apply <code><a href='https://rdrr.io/pkg/DESeq2/man/lfcShrink.html'>DESeq2::lfcShrink()</a></code> if shrunken results are desired.</p></li>
</ul>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Perform pairwise differential expression across groups of cells. Currently
supports edgeR and DESeq2 as DE callers.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>We are no longer recommending the use of software that attempts to
mitigate zero count inflation (e.g. zinbwave, zingeR) for UMI droplet-based
single cell RNA-seq data. Simply model the counts directly.</p>
<p>Updated 2020-02-21.</p>
    <h2 class="hasAnchor" id="deseq-"><a class="anchor" href="#deseq-"></a>DESeq2</h2>

    


<p>We're providing preliminary support for DESeq2 as the differential expression
caller. It is currently considerably slower for large datasets than edgeR.</p>
<p>We're trying to follow the conventions used in DESeq2 for contrasts, defining
the name of the factor in the design formula, numerator, and denominator
level for the fold change calculations. See <code><a href='https://rdrr.io/pkg/DESeq2/man/results.html'>DESeq2::results()</a></code> for details.</p>
<p>Van de Berge and Perraudeau and others have shown the LRT may perform better
for null hypothesis testing, so we use the LRT. In order to use the Wald
test, it is recommended to set <code>useT = TRUE</code> (<em>not currently in use</em>).</p>
<p>For UMI data, for which the expected counts may be very low, the likelihood
ratio test implemented in <code>nbinomLRT()</code> should be used.</p>
<p>Note that DESeq2 supports <code><a href='https://rdrr.io/r/stats/weights.html'>weights()</a></code> values automatically, if slotted using
zinbwave (which is no longer recommended for droplet scRNA-seq).</p>
    <h2 class="hasAnchor" id="edger"><a class="anchor" href="#edger"></a>edgeR</h2>

    


<p>The LRT has been shown to perform better for null hypothesis testing with
droplet scRNA-seq data. Here we are using <code><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>edgeR::glmLRT()</a></code> internally.</p>
<p>edgeR is currently significantly faster than DESeq2 for large datasets.</p>
    <h2 class="hasAnchor" id="seurat-conventions"><a class="anchor" href="#seurat-conventions"></a>Seurat conventions</h2>

    


<p>Note that Seurat currently uses the convention <code>cells.1</code> for the numerator
and <code>cells.2</code> for the denominator. See <code><a href='https://rdrr.io/pkg/Seurat/man/FindMarkers.html'>Seurat::FindMarkers()</a></code> for details.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://rdrr.io/pkg/Seurat/man/WhichCells.html'>Seurat::WhichCells()</a></code>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>Seurat</span>, package <span class='op'>=</span> <span class='st'>"AcidTest"</span><span class='op'>)</span>
<span class='va'>object</span> <span class='op'>&lt;-</span> <span class='va'>Seurat</span>

<span class='co'>## Compare expression in cluster 2 relative to 1.</span>
<span class='va'>clusters</span> <span class='op'>&lt;-</span> <span class='fu'><a href='clusters.html'>clusters</a></span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span>
<span class='va'>numerator</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>clusters</span><span class='op'>)</span><span class='op'>[</span><span class='va'>clusters</span> <span class='op'>==</span> <span class='st'>"2"</span><span class='op'>]</span>
<span class='fu'><a href='summary.html'>summary</a></span><span class='op'>(</span><span class='va'>numerator</span><span class='op'>)</span>
</div><div class='output co'>#&gt;    Length     Class      Mode 
#&gt;        19 character character </div><div class='input'><span class='va'>denominator</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>clusters</span><span class='op'>)</span><span class='op'>[</span><span class='va'>clusters</span> <span class='op'>==</span> <span class='st'>"1"</span><span class='op'>]</span>
<span class='fu'><a href='summary.html'>summary</a></span><span class='op'>(</span><span class='va'>denominator</span><span class='op'>)</span>
</div><div class='output co'>#&gt;    Length     Class      Mode 
#&gt;        25 character character </div><div class='input'>
<span class='co'>## edgeR ====</span>
<span class='co'>## &gt; x &lt;- diffExp(</span>
<span class='co'>## &gt;     object = object,</span>
<span class='co'>## &gt;     numerator = numerator,</span>
<span class='co'>## &gt;     denominator = denominator,</span>
<span class='co'>## &gt;     caller = "edgeR"</span>
<span class='co'>## &gt; )</span>
<span class='co'>## &gt; class(x)</span>
<span class='co'>## &gt; summary(x)</span>

<span class='co'>## DESeq2 ====</span>
<span class='co'>## This will warn about weights with the minimal example.</span>
<span class='co'>## &gt; x &lt;- diffExp(</span>
<span class='co'>## &gt;     object = object,</span>
<span class='co'>## &gt;     numerator = numerator,</span>
<span class='co'>## &gt;     denominator = denominator,</span>
<span class='co'>## &gt;     caller = "DESeq2"</span>
<span class='co'>## &gt; )</span>
<span class='co'>## &gt; class(x)</span>
<span class='co'>## &gt; summary(x)</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


