<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Differential Expression — diffExp • pointillism</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Differential Expression — diffExp" />

<meta property="og:description" content="Perform pairwise differential expression across groups of cells by fitting
to a zero-inflated negative binomial (ZINB) model using the zinbwave package.
Currently supports edgeR and DESeq2 as DE callers." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pointillism</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/steinbaugh/pointillism">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Differential Expression</h1>
    <small class="dont-index">Source: <a href='https://github.com/steinbaugh/pointillism/blob/master/R/AllGenerics.R'><code>R/AllGenerics.R</code></a>, <a href='https://github.com/steinbaugh/pointillism/blob/master/R/diffExp-methods.R'><code>R/diffExp-methods.R</code></a></small>
    <div class="hidden name"><code>diffExp.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Perform pairwise differential expression across groups of cells by fitting
to a zero-inflated negative binomial (ZINB) model using the zinbwave package.
Currently supports edgeR and DESeq2 as DE callers.</p>
    
    </div>

    <pre class="usage"><span class='fu'>diffExp</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='co'># S4 method for SingleCellExperiment</span>
<span class='fu'>diffExp</span>(<span class='no'>object</span>, <span class='no'>numerator</span>, <span class='no'>denominator</span>,
  <span class='kw'>caller</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"edgeR"</span>, <span class='st'>"DESeq2"</span>), <span class='kw'>minCells</span> <span class='kw'>=</span> <span class='fl'>2L</span>, <span class='kw'>minCellsPerGene</span> <span class='kw'>=</span> <span class='fl'>1L</span>,
  <span class='kw'>minCountsPerCell</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>bpparam</span> <span class='kw'>=</span> <span class='kw pkg'>BiocParallel</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/BiocParallel/topics/SerialParam-class'>SerialParam</a></span>())

<span class='co'># S4 method for seurat</span>
<span class='fu'>diffExp</span>(<span class='no'>object</span>, <span class='no'>numerator</span>, <span class='no'>denominator</span>,
  <span class='kw'>caller</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"edgeR"</span>, <span class='st'>"DESeq2"</span>), <span class='kw'>minCells</span> <span class='kw'>=</span> <span class='fl'>2L</span>, <span class='kw'>minCellsPerGene</span> <span class='kw'>=</span> <span class='fl'>1L</span>,
  <span class='kw'>minCountsPerCell</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>bpparam</span> <span class='kw'>=</span> <span class='kw pkg'>BiocParallel</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/BiocParallel/topics/SerialParam-class'>SerialParam</a></span>())</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    <tr>
      <th>numerator</th>
      <td><p><code>character</code>. Cells to use in the numerator of the contrast
(e.g. treatment).</p></td>
    </tr>
    <tr>
      <th>denominator</th>
      <td><p><code>character</code>. Cells to use in the denominator of the
contrast (e.g. control).</p></td>
    </tr>
    <tr>
      <th>caller</th>
      <td><p><code>string</code>. Package to use for differential expression calling.
Defaults to <code>"edgeR"</code> (faster for large datasets) but <code>"DESeq2"</code> is also
supported.</p></td>
    </tr>
    <tr>
      <th>minCells</th>
      <td><p><code>scalar integer</code>. Minimum number of cells required to perform
the differential expression analysis.</p></td>
    </tr>
    <tr>
      <th>minCellsPerGene</th>
      <td><p><code>scalar integer</code>. The minimum number of cells where a
gene is expressed, to pass low expression filtering.</p></td>
    </tr>
    <tr>
      <th>minCountsPerCell</th>
      <td><p><code>scalar integer</code>. Minimum number of counts per cell
for a gene to pass low expression filtering. The number of cells is defined
by <code>minCellsPerGene</code>.</p></td>
    </tr>
    <tr>
      <th>bpparam</th>
      <td><p><code>bpparamClass</code>. BiocParallel parameter specifying the
back-end to be used for computations. See <code><a href='http://www.rdocumentation.org/packages/BiocParallel/topics/register'>BiocParallel::bpparam()</a></code> for
details. We recommend one of the following:</p><ul>
<li><p><code><a href='http://www.rdocumentation.org/packages/BiocParallel/topics/register'>BiocParallel::bpparam()</a></code>.</p></li>
<li><p><code><a href='http://www.rdocumentation.org/packages/BiocParallel/topics/SerialParam-class'>BiocParallel::SerialParam()</a></code>.</p></li>
<li><p><code><a href='http://www.rdocumentation.org/packages/BiocParallel/topics/MulticoreParam-class'>BiocParallel::MulticoreParam()</a></code>.
BiocParallel: https://doi.org/doi:10.18129/B9.bioc.BiocParallel</p></li>
</ul></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Varies depending on the <code>caller</code> argument:</p><ul>
<li><p><code>caller = "edgeR"</code>: <code>DEGLRT</code>.</p></li>
<li><p><code>caller = "DESeq2"</code>: Unshrunken <code>DESeqResults</code>. Use <code>lfcShrink()</code> if
shrunken results are desired.</p></li>
</ul>

    
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>We are currently recommending the ZINB-WaVE method over zingeR, since
it is faster, and has been show to be more sensitive for most single-cell
RNA-seq datasets.</p>
    
    <h2 class="hasAnchor" id="zinbwave"><a class="anchor" href="#zinbwave"></a>zinbwave</h2>

    
    <p>This function will run a lot faster if you pre-calculate the ZINB weights
using the <code><a href='runZinbwave.html'>runZinbwave()</a></code> function, which will stash the weights into the
<code><a href='http://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-assays'>SingleCellExperiment::weights()</a></code> slot of the object. Running zinbwave across
the entire set of filtered cells also has greater sensitivity for weight
calculations.</p>
<p>We are currently using an epsilon setting of <code>1e12</code>, as recommended by the
ZINB-WaVE integration paper. For more information on the zinbwave package,
refer to these materials:</p><ul>
<li><p><a href='https://doi.org/10.1186/s13059-018-1406-4'>zinbwave paper</a>.</p></li>
<li><p><a href='https://bit.ly/2wtDdpS'>zinbwave vignette</a>.</p></li>
<li><p><a href='https://github.com/mikelove/zinbwave-deseq2'>zinbwave-DESeq2 workflow</a>.</p></li>
</ul>
    
    <h2 class="hasAnchor" id="edger"><a class="anchor" href="#edger"></a>edgeR</h2>

    
    <p>After estimation of the dispersions and posterior probabilities, the
<code><a href='http://www.rdocumentation.org/packages/zinbwave/topics/glmWeightedF'>zinbwave::glmWeightedF()</a></code> function is used for statistical inference. This
is an adaptation of <code><a href='http://www.rdocumentation.org/packages/edgeR/topics/glmfit'>edgeR::glmLRT()</a></code>. It uses an F-test for which the
denominator degrees of freedom are by default adjusted according to the
downweighting of excess zeros (<code>ZI = TRUE</code>). Also, independent filtering can
be performed on the obtained p-values (<code>independentFiltering = TRUE</code>). We use
the independent filtering strategy that was originally implemented in DESeq2.
By default, the average fitted values are used as a filter criterion.</p>
    
    <h2 class="hasAnchor" id="deseq-"><a class="anchor" href="#deseq-"></a>DESeq2</h2>

    
    <p>We're providing preliminary support for DESeq2 as the differential expression
caller. It is currently considerably slower for large datasets than edgeR.</p>
<p>We're trying to follow the conventions used in DESeq2 for contrasts, defining
the name of the factor in the design formula, numerator, and denominator
level for the fold change calculations. See <code><a href='http://www.rdocumentation.org/packages/DESeq2/topics/results'>DESeq2::results()</a></code> for more
information.</p>
    
    <h2 class="hasAnchor" id="seurat-conventions"><a class="anchor" href="#seurat-conventions"></a>Seurat conventions</h2>

    
    <p>Note that Seurat currently uses the convention <code>cells.1</code> for the numerator
and <code>cells.2</code> for the denominator. See <code><a href='http://www.rdocumentation.org/packages/Seurat/topics/DiffExpTest'>Seurat::DiffExpTest()</a></code> for
additional information.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='http://www.rdocumentation.org/packages/Seurat/topics/WhichCells'>Seurat::WhichCells()</a></code>.</p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='no'>seurat_small</span>)
<span class='no'>object</span> <span class='kw'>&lt;-</span> <span class='no'>seurat_small</span>

<span class='co'>## Calculate ZINB weights (already stashed).</span>
<span class='co'>## seurat_small &lt;- runZinbwave(object)</span>

<span class='co'>## Compare expression in cluster 3 relative to 2.</span>
<span class='no'>ident</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='clusterID.html'>clusterID</a></span>(<span class='no'>object</span>)
<span class='no'>numerator</span> <span class='kw'>&lt;-</span> <span class='fu'>names</span>(<span class='no'>ident</span>)[<span class='no'>ident</span> <span class='kw'>==</span> <span class='st'>"3"</span>]
<span class='fu'>summary</span>(<span class='no'>numerator</span>)</div><div class='output co'>#&gt;    Length     Class      Mode 
#&gt;        10 character character </div><div class='input'><span class='no'>denominator</span> <span class='kw'>&lt;-</span> <span class='fu'>names</span>(<span class='no'>ident</span>)[<span class='no'>ident</span> <span class='kw'>==</span> <span class='st'>"2"</span>]
<span class='fu'>summary</span>(<span class='no'>denominator</span>)</div><div class='output co'>#&gt;    Length     Class      Mode 
#&gt;        12 character character </div><div class='input'>
<span class='co'>## edgeR.</span>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='fu'>diffExp</span>(
    <span class='kw'>object</span> <span class='kw'>=</span> <span class='no'>object</span>,
    <span class='kw'>numerator</span> <span class='kw'>=</span> <span class='no'>numerator</span>,
    <span class='kw'>denominator</span> <span class='kw'>=</span> <span class='no'>denominator</span>,
    <span class='kw'>caller</span> <span class='kw'>=</span> <span class='st'>"edgeR"</span>
)</div><div class='output co'>#&gt; <span class='message'>Performing differential expression with zinbwave-edgeR</span></div><div class='output co'>#&gt; <span class='message'>Total: 22 cells</span>
#&gt; <span class='message'>Numerator: 10 cells</span>
#&gt; <span class='message'>Denominator: 12 cells</span></div><div class='output co'>#&gt; <span class='message'>Applying gene expression low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>Requiring at least 1 cells with counts of 1 or more per gene</span></div><div class='output co'>#&gt; <span class='message'>187 of 230 genes passed filter</span></div><div class='output co'>#&gt; <span class='message'>Running edgeR.</span></div><div class='output co'>#&gt; <span class='message'>   user  system elapsed </span>
#&gt; <span class='message'>  0.160   0.005   0.168 </span></div><div class='input'><span class='fu'>class</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; [1] "DGELRT"
#&gt; attr(,"package")
#&gt; [1] "edgeR"</div><div class='input'><span class='fu'>summary</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt;                       Length Class            Mode     
#&gt; coefficients           374   -none-           numeric  
#&gt; fitted.values         4114   -none-           numeric  
#&gt; deviance               187   -none-           numeric  
#&gt; method                   1   -none-           character
#&gt; unshrunk.coefficients  374   -none-           numeric  
#&gt; df.residual            187   -none-           numeric  
#&gt; design                  44   -none-           numeric  
#&gt; offset                  22   CompressedMatrix numeric  
#&gt; dispersion             187   -none-           numeric  
#&gt; weights               4114   -none-           numeric  
#&gt; prior.count              1   -none-           numeric  
#&gt; samples                  3   data.frame       list     
#&gt; prior.df                 1   -none-           numeric  
#&gt; AveLogCPM              187   -none-           numeric  
#&gt; df.total               187   -none-           numeric  
#&gt; table                    4   data.frame       list     
#&gt; comparison               1   -none-           character
#&gt; df.test                187   -none-           numeric  </div><div class='input'>
<span class='co'>## DESeq2.</span>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='fu'>diffExp</span>(
    <span class='kw'>object</span> <span class='kw'>=</span> <span class='no'>object</span>,
    <span class='kw'>numerator</span> <span class='kw'>=</span> <span class='no'>numerator</span>,
    <span class='kw'>denominator</span> <span class='kw'>=</span> <span class='no'>denominator</span>,
    <span class='kw'>caller</span> <span class='kw'>=</span> <span class='st'>"DESeq2"</span>
)</div><div class='output co'>#&gt; <span class='message'>Performing differential expression with zinbwave-DESeq2</span></div><div class='output co'>#&gt; <span class='message'>Total: 22 cells</span>
#&gt; <span class='message'>Numerator: 10 cells</span>
#&gt; <span class='message'>Denominator: 12 cells</span></div><div class='output co'>#&gt; <span class='message'>Applying gene expression low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>Requiring at least 1 cells with counts of 1 or more per gene</span></div><div class='output co'>#&gt; <span class='message'>187 of 230 genes passed filter</span></div><div class='output co'>#&gt; <span class='message'>Running DESeq2.</span></div><div class='output co'>#&gt; <span class='message'>converting counts to integer mode</span></div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='warning'>Warning: for 33 row(s), the weights as supplied won't allow parameter estimation, producing a</span>
#&gt; <span class='warning'>  degenerate design matrix. These rows have been flagged in mcols(dds)$weightsFail</span>
#&gt; <span class='warning'>  and treated as if the row contained all zeros (mcols(dds)$allZero set to TRUE).</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='output co'>#&gt; <span class='message'>   user  system elapsed </span>
#&gt; <span class='message'>  1.322   0.022   1.369 </span></div><div class='input'><span class='fu'>class</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; [1] "DESeqResults"
#&gt; attr(,"package")
#&gt; [1] "DESeq2"</div><div class='input'><span class='fu'>summary</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt;       Length        Class         Mode 
#&gt;            6 DESeqResults           S4 </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#note">Note</a></li>

      <li><a href="#zinbwave">zinbwave</a></li>

      <li><a href="#edger">edgeR</a></li>

      <li><a href="#deseq-">DESeq2</a></li>

      <li><a href="#seurat-conventions">Seurat conventions</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='/'>Michael Steinbaugh</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

