<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Find cluster-specific marker genes — findMarkers • pointillism</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Find cluster-specific marker genes — findMarkers" />
<meta property="og:description" content="Find cluster-specific marker genes" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pointillism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../articles/pointillism.html">Get started</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://acidgenomics.com/">
    <span class="fas fa fas fa-vial fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/acidgenomics/pointillism/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Find cluster-specific marker genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/acidgenomics/pointillism/blob/master/R/AllGenerics.R'><code>R/AllGenerics.R</code></a>, <a href='https://github.com/acidgenomics/pointillism/blob/master/R/findMarkers-methods.R'><code>R/findMarkers-methods.R</code></a></small>
    <div class="hidden name"><code>findMarkers.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Find cluster-specific marker genes</p>
    </div>

    <pre class="usage"><span class='fu'>findMarkers</span><span class='op'>(</span><span class='va'>object</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for SingleCellExperiment</span>
<span class='fu'>findMarkers</span><span class='op'>(</span><span class='va'>object</span>, clusters <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for Seurat</span>
<span class='fu'>findMarkers</span><span class='op'>(</span><span class='va'>object</span>, clusters <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>clusters</th>
      <td><p><code>character</code> or <code>NULL</code>.
Cluster identifiers.
Must correspond to values in <code><a href='clusters.html'>clusters()</a></code>.
Note that Seurat uses zero-indexed IDs by default (e.g. 0, 1, 2, ...).
If left <code>NULL</code> (default), all clusters will be analyzed.
Looping across clusters manually here can avoid memory issues on laptops
and other machines with limited amounts of RAM.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Passthrough arguments to <code><a href='diffExp.html'>diffExp()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>list</code> containing:</p><ul>
<li><p><code>caller = "edgeR"</code>: <code>DGELRT</code>.</p></li>
<li><p><code>caller = "DESeq2"</code>: <code>DESeqResults</code>.</p></li>
</ul>

    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Updated 2020-01-30.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>Seurat</span>, package <span class='op'>=</span> <span class='st'>"AcidTest"</span><span class='op'>)</span>

<span class='co'>## Seurat ====</span>
<span class='va'>object</span> <span class='op'>&lt;-</span> <span class='va'>Seurat</span>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>findMarkers</span><span class='op'>(</span><span class='va'>object</span>, caller <span class='op'>=</span> <span class='st'>"edgeR"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'><span style='color: #00BBBB;'>──</span><span> </span><span style='color: #000000;background-color: #BBBBBB;font-weight: bold;'>`findMarkers()`</span><span> </span><span style='color: #00BBBB;'>─────────────────────────────────────────────────────────────</span><span></span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> 3 clusters detected.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Cluster 0</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Performing differential expression with </span><span style='color: #0000BB;'>edgeR</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>    ● Total: 80 cells</span></div><div class='output co'>#&gt; <span class='message'>    ● Numerator: 36 cells</span></div><div class='output co'>#&gt; <span class='message'>    ● Denominator: 44 cells</span></div><div class='output co'>#&gt; <span class='message'>→ Applying gene expression low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>    </span><span style='color: #00BBBB;'>ℹ</span><span> Requiring at least 1 cell with counts of 1 or more per gene.</span></div><div class='output co'>#&gt; <span class='message'>    </span><span style='color: #00BBBB;'>ℹ</span><span> 230 of 230 genes passed filter.</span></div><div class='output co'>#&gt; <span class='message'>→ Applying cell low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>→ Running </span><span style='color: #0000BB;'>edgeR</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Cluster 1</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Performing differential expression with </span><span style='color: #0000BB;'>edgeR</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>    ● Total: 80 cells</span></div><div class='output co'>#&gt; <span class='message'>    ● Numerator: 25 cells</span></div><div class='output co'>#&gt; <span class='message'>    ● Denominator: 55 cells</span></div><div class='output co'>#&gt; <span class='message'>→ Applying gene expression low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>    </span><span style='color: #00BBBB;'>ℹ</span><span> Requiring at least 1 cell with counts of 1 or more per gene.</span></div><div class='output co'>#&gt; <span class='message'>    </span><span style='color: #00BBBB;'>ℹ</span><span> 230 of 230 genes passed filter.</span></div><div class='output co'>#&gt; <span class='message'>→ Applying cell low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>→ Running </span><span style='color: #0000BB;'>edgeR</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Cluster 2</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Performing differential expression with </span><span style='color: #0000BB;'>edgeR</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>    ● Total: 80 cells</span></div><div class='output co'>#&gt; <span class='message'>    ● Numerator: 19 cells</span></div><div class='output co'>#&gt; <span class='message'>    ● Denominator: 61 cells</span></div><div class='output co'>#&gt; <span class='message'>→ Applying gene expression low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>    </span><span style='color: #00BBBB;'>ℹ</span><span> Requiring at least 1 cell with counts of 1 or more per gene.</span></div><div class='output co'>#&gt; <span class='message'>    </span><span style='color: #00BBBB;'>ℹ</span><span> 230 of 230 genes passed filter.</span></div><div class='output co'>#&gt; <span class='message'>→ Applying cell low pass filter.</span></div><div class='output co'>#&gt; <span class='message'>→ Running </span><span style='color: #0000BB;'>edgeR</span><span>.</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/class.html'>class</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "list"</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>class</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $`0`
#&gt; [1] "DGELRT"
#&gt; attr(,"package")
#&gt; [1] "edgeR"
#&gt; 
#&gt; $`1`
#&gt; [1] "DGELRT"
#&gt; attr(,"package")
#&gt; [1] "edgeR"
#&gt; 
#&gt; $`2`
#&gt; [1] "DGELRT"
#&gt; attr(,"package")
#&gt; [1] "edgeR"
#&gt; </div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


