<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot Cell-Type-Specific Gene Markers — plotKnownMarkersDetected • pointillism</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Plot Cell-Type-Specific Gene Markers — plotKnownMarkersDetected" />

<meta property="og:description" content="Plot gene expression per cell in multiple formats:
plotMarkerTSNE(): t-SNE gene expression plot.
plotDot(): Dot plot.
plotViolin(): Violin plot.
" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pointillism</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioSingleCell">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot Cell-Type-Specific Gene Markers</h1>
    <small class="dont-index">Source: <a href='https://github.com/hbc/bcbioSingleCell/blob/master/R/AllGenerics.R'><code>R/AllGenerics.R</code></a>, <a href='https://github.com/hbc/bcbioSingleCell/blob/master/R/plotMarker-methods.R'><code>R/plotMarker-methods.R</code></a></small>
    <div class="hidden name"><code>plotMarker.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Plot gene expression per cell in multiple formats:</p><ol>
<li><p><code>plotMarkerTSNE()</code>: t-SNE gene expression plot.</p></li>
<li><p><code>plotDot()</code>: Dot plot.</p></li>
<li><p><code>plotViolin()</code>: Violin plot.</p></li>
</ol>
    
    </div>

    <pre class="usage"><span class='fu'>plotKnownMarkersDetected</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='fu'>plotMarker</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='fu'>plotTopMarkers</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='co'># S4 method for SingleCellExperiment</span>
<span class='fu'>plotMarker</span>(<span class='no'>object</span>, <span class='no'>genes</span>,
  <span class='kw'>reducedDim</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"TSNE"</span>, <span class='st'>"UMAP"</span>), <span class='kw'>expression</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"mean"</span>, <span class='st'>"median"</span>,
  <span class='st'>"sum"</span>), <span class='kw'>color</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.discrete.color"</span>, <span class='kw'>NULL</span>),
  <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.pointSize"</span>, <span class='fl'>0.75</span>),
  <span class='kw'>pointAlpha</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.pointAlpha"</span>, <span class='fl'>0.8</span>),
  <span class='kw'>pointsAsNumbers</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>label</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.label"</span>, <span class='fl'>TRUE</span>),
  <span class='kw'>labelSize</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.labelSize"</span>, <span class='fl'>6L</span>),
  <span class='kw'>dark</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.dark"</span>, <span class='fl'>FALSE</span>), <span class='kw'>grid</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.grid"</span>,
  <span class='fl'>FALSE</span>), <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.legend"</span>, <span class='fl'>TRUE</span>),
  <span class='kw'>aspectRatio</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"bcbio.aspectRatio"</span>, <span class='fl'>1L</span>), <span class='kw'>title</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='co'># S4 method for SingleCellExperiment</span>
<span class='fu'>plotTopMarkers</span>(<span class='no'>object</span>, <span class='no'>markers</span>, <span class='kw'>n</span> <span class='kw'>=</span> <span class='fl'>10L</span>,
  <span class='kw'>direction</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"positive"</span>, <span class='st'>"negative"</span>, <span class='st'>"both"</span>), <span class='kw'>coding</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>reducedDim</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"TSNE"</span>, <span class='st'>"UMAP"</span>), <span class='kw'>headerLevel</span> <span class='kw'>=</span> <span class='fl'>2L</span>, <span class='no'>...</span>)

<span class='co'># S4 method for SingleCellExperiment</span>
<span class='fu'>plotKnownMarkersDetected</span>(<span class='no'>object</span>, <span class='no'>markers</span>,
  <span class='kw'>reducedDim</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"TSNE"</span>, <span class='st'>"UMAP"</span>), <span class='kw'>headerLevel</span> <span class='kw'>=</span> <span class='fl'>2L</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p><code>character</code>. Gene identifiers. Must match the rownames of the
object.</p></td>
    </tr>
    <tr>
      <th>reducedDim</th>
      <td><p><code>string</code>. Name of reduced dimension matrix slotted in
<code>reducedDims()</code>. Includes TNSE, UMAP, PCA, for example.</p></td>
    </tr>
    <tr>
      <th>expression</th>
      <td><p><code>string</code>. Calculation to apply. Uses <code>match.arg()</code> and
defaults to the first argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>color</th>
      <td><p><code>ggproto</code>/<code>ScaleDiscrete</code> or <code>NULL</code>. Desired ggplot2 color
scale. Must supply discrete values. When set to <code>NULL</code>, the default ggplot2
color palette will be used. If manual color definitions are desired, we
recommend using <code><a href='http://www.rdocumentation.org/packages/ggplot2/topics/scale_manual'>ggplot2::scale_color_manual()</a></code>.
To set the discrete color palette globally, use
<code>options(bcbio.discrete.color = scale_color_viridis_d())</code>.</p></td>
    </tr>
    <tr>
      <th>pointSize</th>
      <td><p><code>scalar numeric</code>. Cell point size.</p></td>
    </tr>
    <tr>
      <th>pointAlpha</th>
      <td><p><code>scalar numeric</code> (<code>0</code>-<code>1</code>). Alpha transparency level.
Useful when there many cells in the dataset, and some cells can be masked.</p></td>
    </tr>
    <tr>
      <th>pointsAsNumbers</th>
      <td><p><code>boolean</code>. Plot the points as numbers (<code>TRUE</code>) or
dots (<code>FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>label</th>
      <td><p><code>boolean</code>. Overlay a cluster identitiy label on the plot.</p></td>
    </tr>
    <tr>
      <th>labelSize</th>
      <td><p><code>scalar integer</code>. Size of the text label.</p></td>
    </tr>
    <tr>
      <th>dark</th>
      <td><p><code>boolean</code>. Plot against a dark background using
<code><a href='http://www.rdocumentation.org/packages/basejump/topics/themes'>basejump::theme_midnight()</a></code>.</p></td>
    </tr>
    <tr>
      <th>grid</th>
      <td><p><code>boolean</code>. Show major grid lines but hide axis lines.</p></td>
    </tr>
    <tr>
      <th>legend</th>
      <td><p><code>boolean</code>. Include plot legend.</p></td>
    </tr>
    <tr>
      <th>aspectRatio</th>
      <td><p><code>scalar integer</code>. Aspect ratio.</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p><code>string</code> or <code>NULL</code>. Plot title.</p></td>
    </tr>
    <tr>
      <th>markers</th>
      <td><p><code>grouped_df</code>. Marker genes data.</p><ul>
<li><p><code>plotTopMarkers()</code>: must be grouped by "<code>cluster</code>".</p></li>
<li><p><code>plotKnownMarkersDetected()</code>: must be grouped by "<code>cellType</code>".</p></li>
</ul></td>
    </tr>
    <tr>
      <th>headerLevel</th>
      <td><p><code>scalar integer</code> (<code>1</code>-<code>7</code>). R Markdown header level.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Show graphical output. Invisibly return <code>ggplot</code> <code>list</code>.</p>
    
    <h2 class="hasAnchor" id="plot-top-markers"><a class="anchor" href="#plot-top-markers"></a>Plot top markers</h2>

    
    <p>The number of markers to plot is determined by the output of the
<code>topMarkers()</code> function. If you want to reduce the number of genes to plot,
simply reassign first using that function. If necessary, we can add support
for the number of genes to plot here in a future update.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other Clustering Functions: <code>cellTypesPerCluster</code>,
  <code>knownMarkersDetected</code>,
  <code>plotCellTypesPerCluster</code>,
  <code>plotFeature</code>, <code>plotPCElbow</code>,
  <code>plotReducedDim</code>,
  <code>sanitizeSeuratMarkers</code>,
  <code>topMarkers</code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># MHC class II genes</span>
<span class='no'>title</span> <span class='kw'>&lt;-</span> <span class='st'>"MHC class II"</span>
<span class='no'>genes</span> <span class='kw'>&lt;-</span> <span class='fu'>rownames</span>(<span class='no'>sce_small</span>)[<span class='fu'>which</span>(<span class='fu'>grepl</span>(
    <span class='kw'>pattern</span> <span class='kw'>=</span> <span class='st'>"^major histocompatibility complex, class II"</span>,
    <span class='kw'>x</span> <span class='kw'>=</span> <span class='fu'>rowData</span>(<span class='no'>sce_small</span>)$<span class='no'>description</span>
))]
<span class='fu'>print</span>(<span class='no'>genes</span>)</div><div class='output co'>#&gt; character(0)</div><div class='input'>
<span class='fu'>plotMarker</span>(<span class='no'>sce_small</span>, <span class='kw'>genes</span> <span class='kw'>=</span> <span class='no'>genes</span>)</div><div class='output co'>#&gt; <span class='error'>Error in assertFormalInterestingGroups(object, value): The interesting groups "sampleName" are not defined as columns in `sampleData()`</span></div><div class='input'><span class='fu'>plotMarker</span>(
    <span class='kw'>object</span> <span class='kw'>=</span> <span class='no'>sce_small</span>,
    <span class='kw'>genes</span> <span class='kw'>=</span> <span class='no'>genes</span>,
    <span class='kw'>expression</span> <span class='kw'>=</span> <span class='st'>"sum"</span>,
    <span class='kw'>pointsAsNumbers</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
    <span class='kw'>dark</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
    <span class='kw'>label</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
    <span class='kw'>title</span> <span class='kw'>=</span> <span class='no'>title</span>
)</div><div class='output co'>#&gt; <span class='error'>Error in assertFormalInterestingGroups(object, value): The interesting groups "sampleName" are not defined as columns in `sampleData()`</span></div><div class='input'>
<span class='co'># Top markers</span>
<span class='no'>markers</span> <span class='kw'>&lt;-</span> <span class='fu'>topMarkers</span>(<span class='no'>all_markers_small</span>, <span class='kw'>n</span> <span class='kw'>=</span> <span class='fl'>1</span>)
<span class='fu'>glimpse</span>(<span class='no'>markers</span>)</div><div class='output co'>#&gt; Observations: 2
#&gt; Variables: 18
#&gt; $ cluster        &lt;fct&gt; 0, 1
#&gt; $ avgLogFC       &lt;dbl&gt; 1.284217, 1.087316
#&gt; $ pct1           &lt;dbl&gt; 0.962, 0.979
#&gt; $ pct2           &lt;dbl&gt; 0.938, 0.865
#&gt; $ rowname        &lt;chr&gt; "SGCE", "IRAK3"
#&gt; $ pvalue         &lt;dbl&gt; 1.642010e-15, 1.291082e-16
#&gt; $ padj           &lt;dbl&gt; 1.642010e-11, 1.291082e-12
#&gt; $ seqnames       &lt;fct&gt; 7, 12
#&gt; $ start          &lt;int&gt; 94524204, 66188879
#&gt; $ end            &lt;int&gt; 94656572, 66254622
#&gt; $ width          &lt;int&gt; 132369, 65744
#&gt; $ strand         &lt;fct&gt; -, +
#&gt; $ broadClass     &lt;fct&gt; coding, coding
#&gt; $ description    &lt;fct&gt; sarcoglycan epsilon [Source:HGNC Symbol;Acc:HGNC:108...
#&gt; $ geneBiotype    &lt;fct&gt; protein_coding, protein_coding
#&gt; $ geneID         &lt;chr&gt; "ENSG00000127990", "ENSG00000090376"
#&gt; $ geneName       &lt;fct&gt; SGCE, IRAK3
#&gt; $ seqCoordSystem &lt;fct&gt; chromosome, chromosome</div><div class='input'><span class='fu'>plotTopMarkers</span>(<span class='no'>seurat_small</span>, <span class='kw'>markers</span> <span class='kw'>=</span> <span class='fu'>tail</span>(<span class='no'>markers</span>, <span class='fl'>1</span>))</div><div class='output co'>#&gt; 
#&gt; 
#&gt; ## Cluster 1 {.tabset}
#&gt; 
#&gt; 
#&gt; 
#&gt; ### IRAK3
#&gt; </div><div class='output co'>#&gt; <span class='error'>Error in assertFormalInterestingGroups(object, value): The interesting groups "sampleName" are not defined as columns in `sampleData()`</span></div><div class='input'>
<span class='co'># Known markers detected</span>
<span class='no'>markers</span> <span class='kw'>&lt;-</span> <span class='fu'>head</span>(<span class='no'>known_markers_small</span>, <span class='kw'>n</span> <span class='kw'>=</span> <span class='fl'>1</span>)
<span class='fu'>glimpse</span>(<span class='no'>markers</span>)</div><div class='output co'>#&gt; Observations: 1
#&gt; Variables: 19
#&gt; $ cellType       &lt;fct&gt; Gut Epithelial Cell
#&gt; $ cluster        &lt;fct&gt; 0
#&gt; $ geneID         &lt;chr&gt; "ENSG00000066405"
#&gt; $ geneName       &lt;chr&gt; "CLDN18"
#&gt; $ avgLogFC       &lt;dbl&gt; -0.9446141
#&gt; $ pct1           &lt;dbl&gt; 0.442
#&gt; $ pct2           &lt;dbl&gt; 0.688
#&gt; $ rowname        &lt;chr&gt; "CLDN18"
#&gt; $ pvalue         &lt;dbl&gt; 3.27838e-06
#&gt; $ padj           &lt;dbl&gt; 0.0327838
#&gt; $ seqnames       &lt;chr&gt; "3"
#&gt; $ start          &lt;int&gt; 137998735
#&gt; $ end            &lt;int&gt; 138033655
#&gt; $ width          &lt;int&gt; 34921
#&gt; $ strand         &lt;chr&gt; "+"
#&gt; $ broadClass     &lt;chr&gt; "coding"
#&gt; $ description    &lt;chr&gt; "claudin 18 [Source:HGNC Symbol;Acc:HGNC:2039]"
#&gt; $ geneBiotype    &lt;chr&gt; "protein_coding"
#&gt; $ seqCoordSystem &lt;chr&gt; "chromosome"</div><div class='input'><span class='fu'>plotKnownMarkersDetected</span>(<span class='no'>seurat_small</span>, <span class='kw'>markers</span> <span class='kw'>=</span> <span class='fu'>head</span>(<span class='no'>markers</span>, <span class='fl'>1</span>))</div><div class='output co'>#&gt; 
#&gt; 
#&gt; ## Gut Epithelial Cell {.tabset}
#&gt; 
#&gt; 
#&gt; 
#&gt; ### CLDN18
#&gt; </div><div class='output co'>#&gt; <span class='error'>Error in assertFormalInterestingGroups(object, value): The interesting groups "sampleName" are not defined as columns in `sampleData()`</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#plot-top-markers">Plot top markers</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Michael Steinbaugh, Rory Kirchner

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='/'>Michael Steinbaugh</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

