<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Sanitize Seurat Markers — SeuratMarkers • pointillism</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Sanitize Seurat Markers — SeuratMarkers" />

<meta property="og:description" content="This generator function is designed to take the original return from a Seurat
marker analysis and add corresponding gene annotations." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pointillism</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/steinbaugh/pointillism">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Sanitize Seurat Markers</h1>
    <small class="dont-index">Source: <a href='https://github.com/steinbaugh/pointillism/blob/master/R/SeuratMarkers.R'><code>R/SeuratMarkers.R</code></a></small>
    <div class="hidden name"><code>SeuratMarkers.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This generator function is designed to take the original return from a Seurat
marker analysis and add corresponding gene annotations.</p>
    
    </div>

    <pre class="usage"><span class='fu'>SeuratMarkers</span>(<span class='no'>markers</span>, <span class='no'>ranges</span>, <span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fl'>0.05</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>markers</th>
      <td><p><code>data.frame</code>. Unmodified <code><a href='http://www.rdocumentation.org/packages/Seurat/topics/FindAllMarkers'>Seurat::FindAllMarkers()</a></code> or
<code><a href='http://www.rdocumentation.org/packages/Seurat/topics/FindMarkers'>Seurat::FindMarkers()</a></code> return.</p></td>
    </tr>
    <tr>
      <th>ranges</th>
      <td><p><code>GRanges</code>. Gene annotations. Names must correspond to the
rownames defined in <code>seurat@data</code>. The function will automatically subset
the ranges and arrange them alphabetically.</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p><code>scalar numeric</code>. Alpha cutoff (adjusted P value; false
discovery rate).</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>SeuratMarkers</code>. Results are arranged by adjusted P value, and
grouped per cluster if applicable.</p>
    
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p><code><a href='http://www.rdocumentation.org/packages/Seurat/topics/FindAllMarkers'>Seurat::FindAllMarkers()</a></code> maps the counts matrix rownames correctly in
the <code>gene</code> column, whereas <code><a href='http://www.rdocumentation.org/packages/Seurat/topics/FindMarkers'>Seurat::FindMarkers()</a></code> maps them correctly in
the rownames of the returned marker <code>data.frame</code>.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='no'>seurat_small</span>)
<span class='no'>object</span> <span class='kw'>&lt;-</span> <span class='no'>seurat_small</span>

<span class='co'>## `FindAllMarkers()` return.</span>
<span class='fu'>invisible</span>(<span class='fu'>capture.output</span>(
    <span class='no'>all_markers</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>Seurat</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/Seurat/topics/FindAllMarkers'>FindAllMarkers</a></span>(<span class='no'>object</span>)
))
<span class='no'>all_sanitized</span> <span class='kw'>&lt;-</span> <span class='fu'>SeuratMarkers</span>(
    <span class='kw'>markers</span> <span class='kw'>=</span> <span class='no'>all_markers</span>,
    <span class='kw'>ranges</span> <span class='kw'>=</span> <span class='fu'>rowRanges</span>(<span class='no'>object</span>)
)</div><div class='output co'>#&gt; <span class='message'>`Seurat::FindAllMarkers()` return detected</span></div><div class='input'><span class='fu'>str</span>(<span class='no'>all_sanitized</span>)</div><div class='output co'>#&gt; Formal class 'SeuratMarkers' [package "pointillism"] with 6 slots
#&gt;   ..@ rownames       : NULL
#&gt;   ..@ nrows          : int 208
#&gt;   ..@ listData       :List of 8
#&gt;   .. ..$ cluster : Factor w/ 4 levels "0","1","2","3": 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. ..$ name    : chr [1:208] "LTB" "TMEM123" "CXCR4" "SRSF7" ...
#&gt;   .. ..$ avgLogFC: num [1:208] 2.5 2.09 1.41 1.71 3.56 ...
#&gt;   .. ..$ pct1    : num [1:208] 1 0.69 0.759 0.759 0.379 0.552 0.034 0.586 0.724 0.207 ...
#&gt;   .. ..$ pct2    : num [1:208] 0.333 0.216 0.275 0.333 0.02 0.137 0.529 0.196 0.392 0.765 ...
#&gt;   .. ..$ pvalue  : num [1:208] 1.54e-11 2.16e-06 3.94e-06 1.14e-05 1.47e-05 ...
#&gt;   .. ..$ padj    : num [1:208] 3.54e-09 4.96e-04 9.05e-04 2.61e-03 3.37e-03 ...
#&gt;   .. ..$ ranges  :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots
#&gt;   .. .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots
#&gt;   .. .. .. .. .. ..@ values         : Factor w/ 273 levels "X","20","1","6",..: 28 8 11 19 8 10 18 5 3 8 ...
#&gt;   .. .. .. .. .. ..@ lengths        : int [1:196] 1 1 2 1 1 1 1 1 1 2 ...
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots
#&gt;   .. .. .. .. .. ..@ start          : int [1:208] 31535744 102267063 136871919 38970741 31677579 118209669 79825597 35852797 186500994 161185024 ...
#&gt;   .. .. .. .. .. ..@ width          : int [1:208] 1968 74053 3817 7896 10942 3791 3686 26909 6696 5466 ...
#&gt;   .. .. .. .. .. ..@ NAMES          : chr [1:208] "LTB" "TMEM123" "CXCR4" "SRSF7" ...
#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots
#&gt;   .. .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": 2 1 2 1 2 1 2 1 2 1 ...
#&gt;   .. .. .. .. .. ..@ lengths        : int [1:97] 7 5 3 4 7 2 7 1 1 2 ...
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots
#&gt;   .. .. .. .. .. ..@ seqnames   : chr [1:273] "X" "20" "1" "6" ...
#&gt;   .. .. .. .. .. ..@ seqlengths : int [1:273] 155270560 63025520 249250621 171115067 198022430 159138663 133851895 135006516 191154276 81195210 ...
#&gt;   .. .. .. .. .. ..@ is_circular: logi [1:273] FALSE FALSE FALSE FALSE FALSE FALSE ...
#&gt;   .. .. .. .. .. ..@ genome     : chr [1:273] "GRCh37" "GRCh37" "GRCh37" "GRCh37" ...
#&gt;   .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package "S4Vectors"] with 6 slots
#&gt;   .. .. .. .. .. ..@ rownames       : NULL
#&gt;   .. .. .. .. .. ..@ nrows          : int 208
#&gt;   .. .. .. .. .. ..@ listData       :List of 10
#&gt;   .. .. .. .. .. .. ..$ broadClass            : Factor w/ 8 levels "coding","ig",..: 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. .. .. .. .. .. ..- attr(*, "names")= chr [1:208] "ENSG00000204487" "ENSG00000152558" "ENSG00000121966" "ENSG00000115875" ...
#&gt;   .. .. .. .. .. .. ..$ entrezID              :List of 208
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204487: int 4050
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000152558: int 114908
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000121966: int 7852
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000115875: int 6432
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100100: int 113791
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000167286: int 915
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000141522: int 396
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000168685: int 3575
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000156976: int [1:6] 100302143 677847 619568 619567 6043 1974
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000158869: int 2207
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000084207: int 2950
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000198851: int 916
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000101439: int 1471
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000164691: int 117289
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000126353: int 1236
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000172005: int 4118
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000177156: int 6888
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000167460: int 7171
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100097: int 3956
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000145220: int 55646
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000170458: int 929
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000137078: int 27240
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000130254: int 9667
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000104921: int 2208
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000025708: int 1890
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000169813: int 3185
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000136732: int 2995
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000072818: int 9744
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163453: int 3490
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100453: int 3002
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000164733: int 1508
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000011600: int 7305
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000143546: int 6279
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163479: int 6746
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000080824: int 3320
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000129277: int 6351
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000168497: int 8436
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163220: int 6280
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000081059: int 6932
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000151500: int 29087
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000111716: int 3945
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000233276: int 2876
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000185201: int 10581
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000025708: int 1890
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000090382: int 4069
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000101439: int 1471
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000170458: int 929
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000084207: int 2950
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000030582: int 2896
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204482: int 7940
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000126759: int 5199
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000110077: int 64231
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163220: int 6280
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000215048: int 3115
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000113269: int 55819
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000161570: int 6352
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000168384: int 3113
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204472: int 199
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000141505: int 432
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197746: int 5660
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000143546: int 6279
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100079: int 3957
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100097: int 3956
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000015475: int 637
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000142089: int 10410
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000162444: int 116362
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000104870: int 2217
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000179639: int 2205
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000085265: int 2219
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204397: int 114769
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204287: int 3122
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163191: int 6282
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100906: int 4792
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000116741: int 5997
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000196126: int [1:4] 100996809 100507714 100507709 3123
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000158869: int 2207
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197249: int 5265
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197405: int 728
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000216490: int 10437
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000145220: int 55646
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000226264: int 3109
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204257: int 3108
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000164733: int 1508
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000173762: int 924
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000101336: int 3055
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000198502: int 3127
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000011600: int 7305
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197766: int 1675
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000171051: int 2357
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000182866: int 3932
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000131981: int 3958
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000166002: int 56935
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000103187: int 23406
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204487: int 4050
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000271614: int 338758
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197629: int 219972
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000109445: int 27309
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000177663: int 23765
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000132514: int 10462
#&gt;   .. .. .. .. .. .. .. .. [list output truncated]
#&gt;   .. .. .. .. .. .. ..$ geneBiotype           : Factor w/ 31 levels "3prime_overlapping_ncrna",..: 19 19 19 19 19 19 19 19 19 19 ...
#&gt;   .. .. .. .. .. .. ..$ geneID                : chr [1:208] "ENSG00000204487" "ENSG00000152558" "ENSG00000121966" "ENSG00000115875" ...
#&gt;   .. .. .. .. .. .. ..$ geneName              : Factor w/ 57063 levels "5S_rRNA","7SK",..: 22792 53320 13547 52079 28806 9704 7224 19742 15019 16133 ...
#&gt;   .. .. .. .. .. .. ..$ seqCoordSystem        : Factor w/ 3 levels "chromosome","lrg",..: 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. .. .. .. .. ..$ gene                  : chr [1:208] "LTB" "TMEM123" "CXCR4" "SRSF7" ...
#&gt;   .. .. .. .. .. .. ..$ gene.mean             : num [1:208] 5.24 3.92 4.54 4.63 3.36 ...
#&gt;   .. .. .. .. .. .. ..$ gene.dispersion       : num [1:208] 6.48 5.1 5.69 7.16 5.77 ...
#&gt;   .. .. .. .. .. .. ..$ gene.dispersion.scaled: num [1:208(1d)] 1.66 -0.553 0.185 1.619 1.125 ...
#&gt;   .. .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 1
#&gt;   .. .. .. .. .. .. .. .. ..$ : chr [1:208] "LTB" "TMEM123" "CXCR4" "SRSF7" ...
#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ elementType    : chr "ANY"
#&gt;   .. .. .. ..@ metadata       :List of 8
#&gt;   .. .. .. .. ..$ organism : chr "Homo sapiens"
#&gt;   .. .. .. .. ..$ ensembldb:'data.frame':	12 obs. of  2 variables:
#&gt;   .. .. .. .. .. ..$ name : chr [1:12] "Db type" "Type of Gene ID" "Supporting package" "Db created by" ...
#&gt;   .. .. .. .. .. ..$ value: chr [1:12] "EnsDb" "Ensembl Gene ID" "ensembldb" "ensembldb package from Bioconductor" ...
#&gt;   .. .. .. .. ..$ build    : chr "GRCh37"
#&gt;   .. .. .. .. ..$ release  : chr "75"
#&gt;   .. .. .. .. ..$ level    : chr "genes"
#&gt;   .. .. .. .. ..$ version  :Classes 'package_version', 'numeric_version'  hidden list of 1
#&gt;   .. .. .. .. .. ..$ : int [1:3] 0 99 0
#&gt;   .. .. .. .. ..$ date     : Date[1:1], format: "2018-10-18"
#&gt;   .. .. .. .. ..$ id       : chr "EnsDb.Hsapiens.v75"
#&gt;   ..@ elementType    : chr "ANY"
#&gt;   ..@ elementMetadata: NULL
#&gt;   ..@ metadata       :List of 4
#&gt;   .. ..$ alpha      : num 0.05
#&gt;   .. ..$ version    :Classes 'package_version', 'numeric_version'  hidden list of 1
#&gt;   .. .. ..$ : int [1:3] 0 2 0
#&gt;   .. ..$ date       : Date[1:1], format: "2018-10-19"
#&gt;   .. ..$ sessionInfo:List of 2
#&gt;   .. .. ..$ platform:List of 9
#&gt;   .. .. .. ..$ version : chr "R version 3.5.1 (2018-07-02)"
#&gt;   .. .. .. ..$ os      : chr "macOS  10.14"
#&gt;   .. .. .. ..$ system  : chr "x86_64, darwin15.6.0"
#&gt;   .. .. .. ..$ ui      : chr "X11"
#&gt;   .. .. .. ..$ language: chr "(EN)"
#&gt;   .. .. .. ..$ collate : chr "en_US.UTF-8"
#&gt;   .. .. .. ..$ ctype   : chr "en_US.UTF-8"
#&gt;   .. .. .. ..$ tz      : chr "America/New_York"
#&gt;   .. .. .. ..$ date    : chr "2018-10-19"
#&gt;   .. .. .. ..- attr(*, "class")= chr "platform_info"
#&gt;   .. .. ..$ packages:Classes 'packages_info' and 'data.frame':	221 obs. of  11 variables:
#&gt;   .. .. .. ..$ package      : chr [1:221] "acepack" "ADGofTest" "annotate" "AnnotationDbi" ...
#&gt;   .. .. .. ..$ ondiskversion: chr [1:221] "1.4.1" "0.3" "1.59.1" "1.43.1" ...
#&gt;   .. .. .. ..$ loadedversion: chr [1:221] "1.4.1" "0.3" "1.59.1" "1.43.1" ...
#&gt;   .. .. .. ..$ path         : chr [1:221] "/Users/mike/R/library/3.5-bioc-3.8-devel/acepack" "/Users/mike/R/library/3.5-bioc-3.8-devel/ADGofTest" "/Users/mike/R/library/3.5-bioc-3.8-devel/annotate" "/Users/mike/R/library/3.5-bioc-3.8-devel/AnnotationDbi" ...
#&gt;   .. .. .. ..$ loadedpath   : chr [1:221] "/Users/mike/R/library/3.5-bioc-3.8-devel/acepack" "/Users/mike/R/library/3.5-bioc-3.8-devel/ADGofTest" "/Users/mike/R/library/3.5-bioc-3.8-devel/annotate" "/Users/mike/R/library/3.5-bioc-3.8-devel/AnnotationDbi" ...
#&gt;   .. .. .. ..$ attached     : logi [1:221] FALSE FALSE FALSE FALSE FALSE FALSE ...
#&gt;   .. .. .. ..$ is_base      : logi [1:221] FALSE FALSE FALSE FALSE FALSE FALSE ...
#&gt;   .. .. .. ..$ date         : chr [1:221] "2016-10-29" "2011-12-28" "2018-07-19" "2018-06-13" ...
#&gt;   .. .. .. ..$ source       : chr [1:221] "CRAN (R 3.5.0)" "CRAN (R 3.5.0)" "Bioconductor" "Bioconductor" ...
#&gt;   .. .. .. ..$ md5ok        : logi [1:221] NA NA NA NA NA NA ...
#&gt;   .. .. .. ..$ library      : Factor w/ 2 levels "/Users/mike/R/library/3.5-bioc-3.8-devel",..: 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. ..- attr(*, "class")= chr "session_info"</div><div class='input'>
<span class='co'>## `FindMarkers()` return.</span>
<span class='fu'>invisible</span>(<span class='fu'>capture.output</span>(
    <span class='no'>ident_3_markers</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>Seurat</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/Seurat/topics/FindMarkers'>FindMarkers</a></span>(
        <span class='kw'>object</span> <span class='kw'>=</span> <span class='no'>object</span>,
        <span class='kw'>ident.1</span> <span class='kw'>=</span> <span class='st'>"1"</span>,
        <span class='kw'>ident.2</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
    )
))
<span class='no'>ident_3_sanitized</span> <span class='kw'>&lt;-</span> <span class='fu'>SeuratMarkers</span>(
    <span class='kw'>markers</span> <span class='kw'>=</span> <span class='no'>ident_3_markers</span>,
    <span class='kw'>ranges</span> <span class='kw'>=</span> <span class='fu'>rowRanges</span>(<span class='no'>object</span>)
)</div><div class='output co'>#&gt; <span class='message'>`Seurat::FindMarkers()` return detected</span></div><div class='input'><span class='fu'>str</span>(<span class='no'>ident_3_sanitized</span>)</div><div class='output co'>#&gt; Formal class 'SeuratMarkers' [package "pointillism"] with 6 slots
#&gt;   ..@ rownames       : NULL
#&gt;   ..@ nrows          : int 196
#&gt;   ..@ listData       :List of 7
#&gt;   .. ..$ name    : chr [1:196] "TYMP" "LYZ" "CST3" "CD14" ...
#&gt;   .. ..$ avgLogFC: num [1:196] 1.77 2.27 1.71 3.19 1.55 ...
#&gt;   .. ..$ pct1    : num [1:196] 0.828 0.862 0.862 0.483 0.862 0.69 0.828 0.621 0.448 0.655 ...
#&gt;   .. ..$ pct2    : num [1:196] 0.196 0.51 0.353 0 0.333 0.118 0.216 0.137 0.059 0.235 ...
#&gt;   .. ..$ pvalue  : num [1:196] 1.13e-08 4.60e-08 6.73e-08 7.13e-08 1.02e-07 ...
#&gt;   .. ..$ padj    : num [1:196] 2.59e-06 1.06e-05 1.55e-05 1.64e-05 2.35e-05 ...
#&gt;   .. ..$ ranges  :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots
#&gt;   .. .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots
#&gt;   .. .. .. .. .. ..@ values         : Factor w/ 273 levels "X","20","1","6",..: 19 7 2 18 8 10 4 1 8 3 ...
#&gt;   .. .. .. .. .. ..@ lengths        : int [1:185] 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots
#&gt;   .. .. .. .. .. ..@ start          : int [1:196] 50964181 69742121 23608534 140011313 67351066 42422614 31553901 47483612 59939081 153330330 ...
#&gt;   .. .. .. .. .. ..@ width          : int [1:196] 4305 5894 10577 1974 3066 7857 2786 6093 13059 3174 ...
#&gt;   .. .. .. .. .. ..@ NAMES          : chr [1:196] "TYMP" "LYZ" "CST3" "CD14" ...
#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots
#&gt;   .. .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": 2 1 2 1 2 1 2 1 2 1 ...
#&gt;   .. .. .. .. .. ..@ lengths        : int [1:103] 1 1 2 3 2 2 3 1 4 1 ...
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots
#&gt;   .. .. .. .. .. ..@ seqnames   : chr [1:273] "X" "20" "1" "6" ...
#&gt;   .. .. .. .. .. ..@ seqlengths : int [1:273] 155270560 63025520 249250621 171115067 198022430 159138663 133851895 135006516 191154276 81195210 ...
#&gt;   .. .. .. .. .. ..@ is_circular: logi [1:273] FALSE FALSE FALSE FALSE FALSE FALSE ...
#&gt;   .. .. .. .. .. ..@ genome     : chr [1:273] "GRCh37" "GRCh37" "GRCh37" "GRCh37" ...
#&gt;   .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package "S4Vectors"] with 6 slots
#&gt;   .. .. .. .. .. ..@ rownames       : NULL
#&gt;   .. .. .. .. .. ..@ nrows          : int 196
#&gt;   .. .. .. .. .. ..@ listData       :List of 10
#&gt;   .. .. .. .. .. .. ..$ broadClass            : Factor w/ 8 levels "coding","ig",..: 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. .. .. .. .. .. ..- attr(*, "names")= chr [1:196] "ENSG00000025708" "ENSG00000090382" "ENSG00000101439" "ENSG00000170458" ...
#&gt;   .. .. .. .. .. .. ..$ entrezID              :List of 196
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000025708: int 1890
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000090382: int 4069
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000101439: int 1471
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000170458: int 929
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000084207: int 2950
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000030582: int 2896
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204482: int 7940
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000126759: int 5199
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000110077: int 64231
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163220: int 6280
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000215048: int 3115
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000113269: int 55819
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000161570: int 6352
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000168384: int 3113
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204472: int 199
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000141505: int 432
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197746: int 5660
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000143546: int 6279
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100079: int 3957
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100097: int 3956
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000015475: int 637
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000142089: int 10410
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000162444: int 116362
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000104870: int 2217
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000179639: int 2205
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000085265: int 2219
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204397: int 114769
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204287: int 3122
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163191: int 6282
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100906: int 4792
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000116741: int 5997
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000196126: int [1:4] 100996809 100507714 100507709 3123
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000158869: int 2207
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197249: int 5265
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197405: int 728
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000216490: int 10437
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000145220: int 55646
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000226264: int 3109
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204257: int 3108
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000164733: int 1508
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000173762: int 924
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000101336: int 3055
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000198502: int 3127
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000011600: int 7305
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197766: int 1675
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000171051: int 2357
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000182866: int 3932
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000131981: int 3958
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000166002: int 56935
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000103187: int 23406
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204487: int 4050
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000271614: int 338758
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197629: int 219972
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000109445: int 27309
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000177663: int 23765
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000132514: int 10462
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000175489: int 126364
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000126883: int 8021
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000130066: int 6303
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163131: int 1520
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000129226: int 968
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100300: int 706
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000167286: int 915
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000185896: int 3916
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000168961: int 3965
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000133321: int 5920
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000145649: int 3001
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000130489: int 9997
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000113140: int 6678
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000180573: int 8334
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000112799: int 9450
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000177156: int 6888
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000090104: int 5996
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000125538: int 3553
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000120885: int 1191
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000127920: int 2791
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000008517: int 9235
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000179344: int [1:2] 101060835 3119
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000072818: int 9744
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000163736: int 5473
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000166681: int 27018
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000005961: int 3674
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000139187: int 10219
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000169704: int 2815
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000101162: int 81027
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000182718: int 302
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000198821: int 919
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000180644: int 5551
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000169583: int 9022
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000161911: int 340205
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100385: int 3560
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000101856: int 10857
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000196139: int 8644
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000100100: int 113791
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000136732: int 2995
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000204276: int 3118
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000196735: int [1:2] 100509457 3117
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000168497: int 8436
#&gt;   .. .. .. .. .. .. .. ..$ ENSG00000197540: int 3004
#&gt;   .. .. .. .. .. .. .. .. [list output truncated]
#&gt;   .. .. .. .. .. .. ..$ geneBiotype           : Factor w/ 31 levels "3prime_overlapping_ncrna",..: 19 19 19 19 19 19 19 19 19 19 ...
#&gt;   .. .. .. .. .. .. ..$ geneID                : chr [1:196] "ENSG00000025708" "ENSG00000090382" "ENSG00000101439" "ENSG00000170458" ...
#&gt;   .. .. .. .. .. .. ..$ geneName              : Factor w/ 57063 levels "5S_rRNA","7SK",..: 54729 22870 11204 9653 17745 17598 22788 10111 25310 49699 ...
#&gt;   .. .. .. .. .. .. ..$ seqCoordSystem        : Factor w/ 3 levels "chromosome","lrg",..: 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. .. .. .. .. ..$ gene                  : chr [1:196] "TYMP" "LYZ" "CST3" "CD14" ...
#&gt;   .. .. .. .. .. .. ..$ gene.mean             : num [1:196] 3.85 5.97 5.59 2.25 4.01 ...
#&gt;   .. .. .. .. .. .. ..$ gene.dispersion       : num [1:196] 4.52 6.8 6.24 4.34 4.51 ...
#&gt;   .. .. .. .. .. .. ..$ gene.dispersion.scaled: num [1:196(1d)] -1.46 0.707 -0.707 -0.518 -1.304 ...
#&gt;   .. .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 1
#&gt;   .. .. .. .. .. .. .. .. ..$ : chr [1:196] "TYMP" "LYZ" "CST3" "CD14" ...
#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"
#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL
#&gt;   .. .. .. .. .. ..@ metadata       : list()
#&gt;   .. .. .. ..@ elementType    : chr "ANY"
#&gt;   .. .. .. ..@ metadata       :List of 8
#&gt;   .. .. .. .. ..$ organism : chr "Homo sapiens"
#&gt;   .. .. .. .. ..$ ensembldb:'data.frame':	12 obs. of  2 variables:
#&gt;   .. .. .. .. .. ..$ name : chr [1:12] "Db type" "Type of Gene ID" "Supporting package" "Db created by" ...
#&gt;   .. .. .. .. .. ..$ value: chr [1:12] "EnsDb" "Ensembl Gene ID" "ensembldb" "ensembldb package from Bioconductor" ...
#&gt;   .. .. .. .. ..$ build    : chr "GRCh37"
#&gt;   .. .. .. .. ..$ release  : chr "75"
#&gt;   .. .. .. .. ..$ level    : chr "genes"
#&gt;   .. .. .. .. ..$ version  :Classes 'package_version', 'numeric_version'  hidden list of 1
#&gt;   .. .. .. .. .. ..$ : int [1:3] 0 99 0
#&gt;   .. .. .. .. ..$ date     : Date[1:1], format: "2018-10-18"
#&gt;   .. .. .. .. ..$ id       : chr "EnsDb.Hsapiens.v75"
#&gt;   ..@ elementType    : chr "ANY"
#&gt;   ..@ elementMetadata: NULL
#&gt;   ..@ metadata       :List of 4
#&gt;   .. ..$ alpha      : num 0.05
#&gt;   .. ..$ version    :Classes 'package_version', 'numeric_version'  hidden list of 1
#&gt;   .. .. ..$ : int [1:3] 0 2 0
#&gt;   .. ..$ date       : Date[1:1], format: "2018-10-19"
#&gt;   .. ..$ sessionInfo:List of 2
#&gt;   .. .. ..$ platform:List of 9
#&gt;   .. .. .. ..$ version : chr "R version 3.5.1 (2018-07-02)"
#&gt;   .. .. .. ..$ os      : chr "macOS  10.14"
#&gt;   .. .. .. ..$ system  : chr "x86_64, darwin15.6.0"
#&gt;   .. .. .. ..$ ui      : chr "X11"
#&gt;   .. .. .. ..$ language: chr "(EN)"
#&gt;   .. .. .. ..$ collate : chr "en_US.UTF-8"
#&gt;   .. .. .. ..$ ctype   : chr "en_US.UTF-8"
#&gt;   .. .. .. ..$ tz      : chr "America/New_York"
#&gt;   .. .. .. ..$ date    : chr "2018-10-19"
#&gt;   .. .. .. ..- attr(*, "class")= chr "platform_info"
#&gt;   .. .. ..$ packages:Classes 'packages_info' and 'data.frame':	221 obs. of  11 variables:
#&gt;   .. .. .. ..$ package      : chr [1:221] "acepack" "ADGofTest" "annotate" "AnnotationDbi" ...
#&gt;   .. .. .. ..$ ondiskversion: chr [1:221] "1.4.1" "0.3" "1.59.1" "1.43.1" ...
#&gt;   .. .. .. ..$ loadedversion: chr [1:221] "1.4.1" "0.3" "1.59.1" "1.43.1" ...
#&gt;   .. .. .. ..$ path         : chr [1:221] "/Users/mike/R/library/3.5-bioc-3.8-devel/acepack" "/Users/mike/R/library/3.5-bioc-3.8-devel/ADGofTest" "/Users/mike/R/library/3.5-bioc-3.8-devel/annotate" "/Users/mike/R/library/3.5-bioc-3.8-devel/AnnotationDbi" ...
#&gt;   .. .. .. ..$ loadedpath   : chr [1:221] "/Users/mike/R/library/3.5-bioc-3.8-devel/acepack" "/Users/mike/R/library/3.5-bioc-3.8-devel/ADGofTest" "/Users/mike/R/library/3.5-bioc-3.8-devel/annotate" "/Users/mike/R/library/3.5-bioc-3.8-devel/AnnotationDbi" ...
#&gt;   .. .. .. ..$ attached     : logi [1:221] FALSE FALSE FALSE FALSE FALSE FALSE ...
#&gt;   .. .. .. ..$ is_base      : logi [1:221] FALSE FALSE FALSE FALSE FALSE FALSE ...
#&gt;   .. .. .. ..$ date         : chr [1:221] "2016-10-29" "2011-12-28" "2018-07-19" "2018-06-13" ...
#&gt;   .. .. .. ..$ source       : chr [1:221] "CRAN (R 3.5.0)" "CRAN (R 3.5.0)" "Bioconductor" "Bioconductor" ...
#&gt;   .. .. .. ..$ md5ok        : logi [1:221] NA NA NA NA NA NA ...
#&gt;   .. .. .. ..$ library      : Factor w/ 2 levels "/Users/mike/R/library/3.5-bioc-3.8-devel",..: 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. ..- attr(*, "class")= chr "session_info"</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#note">Note</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='/'>Michael Steinbaugh</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

